joomla/joomla-cms
# [4.0] Upgrade inital pw length to 12 characters without complexitiy rules

Pull Request for Issue #29476

### Summary of Changes

As mentioned in a bunch of tickets (see https://github.com/joomla/joomla-cms/issues/29476), we have either nor password policy at all or a really bad one.

Therefore, we the JSST suggest the following for the Joomla Core:
* enforce a password policy in Joomla 4 during installation
* update the default settings of the password policy settings in com_users with the same policy

Policy would be:
* minimum 12 characters
* no other complexity rules

### Testing Instructions

- install this branch as new install
- choose an PW that is lower than 12 chars
- notice the message about the PW beeing to short.
- (please ignore the message about Mysql 8 as it is not realted to this changes and just come up when any kind of validation fails; this has to be fixed in a different PR)
- try to create an user from the frontend
- notice the minimum character count message
- try to create an user from the backend
- notice the minimum character count message

### Actual result BEFORE applying this Pull Request

by default in the installer we have no PW policy
by default we had the min 4 char PW policy.

### Expected result AFTER applying this Pull Request

by default we have a 12 character minimum lenth pw policy enfored in the installer and set per default 

### Documentation Changes Required

None.

### Other changes

- move language strings from com_users to the main as this is a field that should not depend on com_users strings
- upgrade the auto generated PW to 32 chars
- make the messages showing up as error given that the validation fails ;)

cc @joomla/security
